modules/gamoshiBidAdapter.js

```diff
@@ -54,25 +58,25 @@ export const spec = {
   buildRequests: function (validBidRequests, bidderRequest) {
     return validBidRequests.map(bidRequest => {
       const {adUnitCode, auctionId, mediaTypes, params, sizes, transactionId} = bidRequest;
-      const baseEndpoint = params['rtbEndpoint'] || 'https://rtb.gamoshi.io';
+      const baseEndpoint = params['rtbEndpoint'] || ENDPOINTS[bidRequest.bidder] || ENDPOINTS['gamoshi'];
       const rtbEndpoint = `${baseEndpoint}/r/${params.supplyPartnerId}/bidr?rformat=open_rtb&reqformat=rtb_json&bidder=prebid` + (params.query ? '&' + params.query : '');
-      let referer = bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.referer;
-      let url = config.getConfig('pageUrl') || referer || utils.getTopWindowUrl();
+      let url = config.getConfig('pageUrl') || bidderRequest.refererInfo.referer;
 
       const rtbBidRequest = {
         'id': auctionId,
         'site': {
           'domain': helper.getTopWindowDomain(url),
           'page': url,
-          'ref': helper.getTopWindowReferer()
+          'ref': bidderRequest.refererInfo.referer
         },
         'device': {
           'ua': navigator.userAgent
         },
         'imp': [],
         'ext': {}
       };
-      if (bidderRequest && bidderRequest.gdprConsent) {
+
+      if (bidderRequest.gdprConsent) {
```

Comment:
If a pub enables the consent module, but for some reason the consent information cannot be found,
then prebid passes an object like so to the bidders:

```javascript
{
   gdprConsent: {
      consentString: undefined
      gdprApplies: undefined
      vendorData: undefined
  }
}
```

which in this scenario would end up sending an empty object to your exchange like so

```javascript
{
   ... 
   ext: {
      gdpConsent: {}
   }
}
```

Looks like your ad server handles it just fine but wanted to make it known!

Reply:
Thanks for the insight, we will update the code in the next PR.